app/services/support_interface/interview_changes_export.rb

```diff
@@ -0,0 +1,93 @@
+module SupportInterface
+  class InterviewChangesExport
+    def data_for_export
+      interview_audits.map do |audit|
+        row_for_audit(audit)
+      end
+    end
+
+    def row_for_audit(audit)
+      interview = audit.auditable
+      application_choice = interview.application_choice
+      {
+        audit_id: audit.id,
+        audit_created_at: audit.created_at,
+        audit_type: audit.action,
+        interview_id: interview.id,
+        candidate_id: application_choice.candidate.id,
+        application_form_id: application_choice.application_form.id,
+        provider_code: application_choice.provider.code,
+        provider_user: audit_user(audit),
+        interview_status: interview_status(interview, audit),
+      }.merge(
+        interview_change(audit).transform_keys(&:to_sym),
+      ).merge(
+        {
+          interview_preferences: application_choice.application_form.interview_preferences,
+          application_submitted_at: application_choice.application_form.submitted_at,
+          course_code: application_choice.course.code,
+          course_location: application_choice.site.name_and_code,
+        },
+      )
+    end
+
+  private
+
+    def interview_audits
+      Audited::Audit.where(auditable_type: 'Interview')
+    end
+
+    def interview_status(interview, audit)
+      if audit.created_at < interview_time_when_change_made(interview, audit)
```

Comment:
I found out today that we have access to `audit.revision` which will give us the interview at it's state at that time, so maybe we can use this instead:
```
if audit.created_at < audit.revision.date_and_time
```

Reply:
Oh this is a nice feature, TIL!

Reply:
Looking at the source for this, it will go through the entire audit history up to the point of this audit to reconstruct the object. That is less efficient than my method, but perhaps more readable. Given this is an export and any performance issues would only affect support users, shall we go with the readable one?

Reply:
spoke in person but for posterity: we're going to avoid revision for now as there is more work to be done to investigate building a performant solution.